comid-triples.cddl

```diff
@@ -0,0 +1,142 @@
+concise-mid-tag = {
+  ; TODO(tho) language
+  ; TODO(tho) tag-id
+  ; TODO(tho) module-name
+  ; TODO(tho) entity
+  ; TODO(tho) linked-tags
+  comid.claims => claims-map
+  ; TODO(tho) extension
+}
+
+claims-map = non-empty<{
+  comid.reference-claims => one-or-more<reference-claim-record>
+  comid.endorsed-claims => one-or-more<endorsed-claim-record>
+  comid.identity-claims => one-or-more<identity-claim-record>
+  * $$claims-map-extension
+}>
+
+; REFERENCE-VALUE triple
+; "target environment $TE" "has reference measurements" "$RV"
+reference-claim-record = {
+  target-environment: environment-map 
+  reference-values: values-map
+}
+
+; ENDORSED-VALUE triple
+; "environment $E" "has endorsed measurements" "$EV"
+endorsed-claim-record = {
+  environment: environment-map 
+  endorsed-values: values-map
+}
+
+; DEVICE-IDENTITY triple
+; (Should it be AE-IDENTITY instead?)
+; "attesting environment $AE" "signs Evidence with key" "$K"
+identity-claim-record = {
+  attesting-environment: environment-map 
+  identity: COSE_Key
+}
+
+environment-map = non-empty<{
+  ? comid.class => class-map
+  ? comid.instance => instance-map
+}>
+
+class-map = {
+  comid.class-id => $class-id-type-choice
+  ; TODO(tho) add layer, index, etc. or add them as
+  ; class-map-extension?
+  * $$class-map-extension
+}
+
+$class-id-type-choice /= tagged-oid-type
+$class-id-type-choice /= tagged-uuid-type
+$class-id-type-choice /= tagged-impl-id-type
+
+instance-map = {
+  comid.instance-id => $instance-id-type-choice
+  * $$instance-map-extension
+}
```

Comment:
Should we also not have instance identifiers, like MAC addresses or IP Addresses belonging to specific instances?

Reply:
for MAC addresses, this is already absorbed by tagged-ueid-type (0x02 IEEE EUI).  IP addresses are locators, not identifiers, so I'd really push back against anyone trying to use them that way.

Reply:
My suggestion is that we consider each member of the instance-value-group-choice to see if it makes sense as an instance-id. If yes, then allow it in both places (as an environment identifier and as a measurement).